pkg/network/dhcp/configurator_test.go

```diff
@@ -38,13 +38,13 @@ var _ = Describe("DHCP configurator", func() {
 
 	newConfiguratorWithClientFilter := func(launcherPID string, advertisingIfaceName string) *Configurator {
 		configurator := NewConfiguratorWithClientFilter(fake.NewFakeInMemoryNetworkCacheFactory(), launcherPID, advertisingIfaceName, netdriver.NewMockNetworkHandler(gomock.NewController(GinkgoT())))
```

Comment:
Why don't we mock the whole configurator? Same as done for the infra configurator one?

Reply:
this is the real deal we want to test, we cannot mock it.

Reply:
Why podNIC should test the internal logic of the dhcp configurator?

Reply:
oh, here we are in the configurator_test. Ignore my comment.